api/redirects.go

```diff
@@ -153,7 +174,122 @@ func isValidRelativePath(path string) bool {
 	return strings.HasPrefix(path, "/") && !strings.HasPrefix(path, "//")
 }
 
-func isValidBase64(s string) bool {
-	_, err := base64.StdEncoding.DecodeString(s)
-	return err == nil
+// encodeBase64 returns the base64 encoded string of the original URL key string
+func encodeBase64(key string) string {
+	encodedKey := base64.StdEncoding.EncodeToString([]byte(key))
+
+	return encodedKey
+}
+
+// getRedirects gets a paged list of redirects from the store
+func (api *RedirectAPI) getRedirects(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	strCount := req.URL.Query().Get("count")
+	strCursor := req.URL.Query().Get("cursor")
+
+	// make count default to 10
+	if strCount == "" {
+		strCount = "10"
+	}
+
+	// make cursor default to 0
+	if strCursor == "" {
+		strCursor = "0"
+	}
+
+	logData := log.Data{"count": strCount, "cursor": strCursor}
+
+	// convert count from a string to an int64 ready to use in call to GetRedirects
+	count, errCount := strconv.ParseInt(strCount, 10, 32)
+
+	if errCount != nil {
+		log.Info(ctx, "invalid path parameter - failed to convert count to int64", logData)
+		api.handleError(ctx, w, ErrInvalidCount, http.StatusBadRequest)
+		return
+	}
+
+	// validate count
+	if count < 0 {
+		log.Info(ctx, "invalid path parameter - count should be a positive integer", logData)
+		api.handleError(ctx, w, ErrNegativeCount, http.StatusBadRequest)
+		return
+	}
+
+	// convert cursor from a string to a uint64 ready to use in call to GetRedirects
+	cursor, err := strconv.ParseUint(strCursor, 10, 32)
+	if err != nil {
+		log.Info(ctx, "invalid path parameter - failed to convert cursor to uint64", logData)
+		api.handleError(ctx, w, ErrInvalidOrNegativeCursor, http.StatusBadRequest)
+		return
+	}
+	logData = log.Data{"count": count, "cursor": cursor}
+
+	keyValuePairs, newCursor, err := api.RedirectStore.GetRedirects(ctx, count, cursor)
+	if err != nil {
+		log.Error(ctx, "redis failed on getting redirects", err, logData)
+		api.handleError(ctx, w, ErrInternal, http.StatusInternalServerError)
+		return
+	}
+
+	logData = log.Data{"num_redirects": len(keyValuePairs)}
+	log.Info(ctx, "redirects retrieved from redis", logData)
+
+	redirectList := make([]models.Redirect, 0, len(keyValuePairs))
+
+	redirectLinkBuilder := links.FromHeadersOrDefault(&req.Header, api.apiURL)
```

Comment:
Rename the link builder as it's generic and can be reused should we need to add any additional links in future.
```suggestion
	linkBuilder := links.FromHeadersOrDefault(&req.Header, api.apiURL)
```